src/hooks/use-sorting/use-sorting.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import sortBy from 'lodash/sortBy';
+
+const defaultSortMethod = (items, column) => sortBy(items, column);
+
+const sortItems = (
+  items,
+  column,
+  direction,
+  sortMethod = defaultSortMethod
+) => {
+  if (!column) {
+    return items;
+  }
+
+  const sortedItems = sortMethod(items, column);
+
+  if (direction === 'desc') {
+    return sortedItems.reverse();
+  }
+
+  return sortedItems;
+};
+
+const useSorting = (items, column, sortDirection, sortMethod) => {
+  const initialState = {
```

Comment:
Nit: this could live outside the hook as `const getInitialState = ({...}) =>` then inside the hook we cna pass the function to React.useState. It will only be invoked once. The fn avoids a redefinition of the obejct and a `sortItems` invocation which is then actually not used by the hook anymore.

Additionally we could send the state ourside the hook itself. So:

```js
const useSortingState = () => {
   const [sortState, setSorting] = React.useState(getInitialSortingState);
   
   return [sortState, setSorting]; 
}
```

This allow us to `useDebugValue`

```js
const useSortingState = () => {
   const [sortState, setSorting] = React.useState(getInitialSortingState);
   
   React.useDebugValue({
      sortState
   })

   return [sortState, setSorting]; 
}
```

which can be helpful for consumers in the apps as it shows up nicely in the React DevTools.

Reply:
This is interesting! Thanks, I'll try it